src/contract/error.rs

```diff
@@ -0,0 +1,38 @@
+//! Contract call/query error.
+
+#![allow(unknown_lints)]
+#![allow(missing_docs)]
+
+use ethabi;
+
+error_chain! {
+  links {
+    Abi(ethabi::Error, ethabi::ErrorKind);
+    Api(::Error, ::ErrorKind);
```

Comment:
isn't it better to use `foreign_links` here?

Reply:
`links` are used for other `error_chains`.